galaxy_ng/app/api/ui/viewsets/collection.py

```diff
@@ -127,39 +129,36 @@ class CollectionVersionViewSet(api_base.GenericViewSet):
     serializer_class = serializers.CollectionVersionSerializer
 
     def list(self, request, *args, **kwargs):
-        self.paginator.init_from_request(request)
-
-        params = self.request.query_params.dict()
-        params.update({
-            'offset': self.paginator.offset,
-            'limit': self.paginator.limit,
-        })
-
-        # pulp uses ordering, but the UI is standardized around sort
-        if params.get('sort'):
-            params['ordering'] = params.get('sort')
-            del params['sort']
+        queryset = CollectionVersion.objects.all()
+        repo = self.request.query_params.dict().get('repository')
+        if repo:
+            try:
+                distro = AnsibleDistribution.objects.get(base_path=repo)
+            except ObjectDoesNotExist:
+                raise NotFound(f'Repository not found: {repo}')
+            repository_version = distro.repository.latest_version()
+            queryset = queryset.filter(pk__in=repository_version.content)
 
-        api = galaxy_pulp.PulpCollectionsApi(pulp.get_client())
-        response = api.list(exclude_fields='docs_blob', **params)
-
-        data = serializers.CollectionVersionSerializer(response.results, many=True).data
-        return self.paginator.paginate_proxy_response(data, response.count)
+        page = self.paginate_queryset(queryset)
+        serializer = self.get_serializer(page, many=True)
+        return self.get_paginated_response(serializer.data)
 
     @swagger_auto_schema(operation_summary="Retrieve collection version",
                          responses={200: serializers.CollectionVersionDetailSerializer})
     def retrieve(self, request, *args, **kwargs):
         namespace, name, version = self.kwargs['version'].split('/')
-
-        api = galaxy_pulp.PulpCollectionsApi(pulp.get_client())
-        response = api.list(namespace=namespace, name=name, version=version, limit=1)
-
-        if not response.results:
-            raise NotFound()
-
-        data = serializers.CollectionVersionDetailSerializer(response.results[0]).data
-        return Response(data)
-
+        try:
```

Comment:
Could this use django.shortcut.get_object_or_404() ?

(Though, the error messages created by get_object_or_404() aren't great... maybe we should add a version of get_object_or_404 that generates better errors and use it where the try/except ObjectDoesNotExist are?)

Reply:
Yeah that is why I didn't use it because I couldn't add details about the object in the error message, but since it is a GET on the url params maybe the 404 without object info in the error message is ok